From 14aaa7b6ca11f9580e0a4b424716b265e292b35a Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 30 Aug 2022 15:58:32 -0700
Subject: [PATCH] Add needed headers for including mbstate_t and exit()

Upstream-Status: Inappropriate [Reconfigure will solve it]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index c2462f7f..33668cf0 100755
--- a/configure
+++ b/configure
@@ -3458,6 +3458,7 @@ rm -f "conftest.$ac_objext" "conftest.$ac_ext"
   cat >"conftest.$ac_ext" <<_ACEOF
 #line 3459 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 $ac_declaration
 int
 main (void)
@@ -13526,6 +13527,7 @@ cat >"conftest.$ac_ext" <<_ACEOF
 #include <stdlib.h>
 #include <stdarg.h>
 #include <stdio.h>
+#include <wchar.h>
 #ifdef HAVE_LIBUTF8_H
 #include <libutf8.h>
 #endif
