From 3d3d09df34d54cb6904715cdb21fce085d5c225a Mon Sep 17 00:00:00 2001
From: Michael Scott <michael.scott@linaro.org>
Date: Mon, 24 Apr 2017 11:06:32 -0700
Subject: [PATCH] fix PARTITION_UUIDS for mx6cuboxi

Signed-off-by: Michael Scott <michael.scott@linaro.org>
---
 include/configs/mx6cuboxi.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/configs/mx6cuboxi.h b/include/configs/mx6cuboxi.h
index 8132b1c..6985599 100644
--- a/include/configs/mx6cuboxi.h
+++ b/include/configs/mx6cuboxi.h
@@ -70,7 +70,6 @@
 #define CONFIG_SYS_FSL_USDHC_NUM	1
 #define CONFIG_SYS_MMC_ENV_DEV		0	/* SDHC2 */
 
-#define CONFIG_PARTITION_UUIDS
 #define CONFIG_CMD_PART
 
 #define CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
