From a5689c8958122d364d950db9514128e36a07d8ab Mon Sep 17 00:00:00 2001
From: Tyler Baker <tyler@foundries.io>
Date: Fri, 8 Mar 2019 10:24:33 -0800
Subject: [PATCH] apalis_imx6 run distro bootcmd by default

Signed-off-by: Tyler Baker <tyler@foundries.io>
---
 include/configs/apalis_imx6.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/apalis_imx6.h b/include/configs/apalis_imx6.h
index 71dfeb56f5..067b990525 100644
--- a/include/configs/apalis_imx6.h
+++ b/include/configs/apalis_imx6.h
@@ -180,6 +180,7 @@
         func(DHCP, dhcp, na)
 #include <config_distro_bootcmd.h>
 #include <config_distro_defaults.h>
+#define CONFIG_BOOTCOMMAND "setenv fdtfile imx6q-apalis-ixora-v1.1.dtb && run distro_bootcmd;"
 #undef CONFIG_ISO_PARTITION
 #else
 #define BOOTENV
-- 
2.17.2 (Apple Git-113)

