diff --git a/src/import/vendor.fetch/github.com/docker/cli/cli/config/config.go b/src/import/vendor.fetch/github.com/docker/cli/cli/config/config.go
index 31ad117d41..eaf851e22c 100644
--- a/src/import/vendor.fetch/github.com/docker/cli/cli/config/config.go
+++ b/src/import/vendor.fetch/github.com/docker/cli/cli/config/config.go
@@ -21,6 +21,7 @@ const (
 	configFileDir  = ".docker"
 	oldConfigfile  = ".dockercfg"
 	contextsDir    = "contexts"
+	defaultSystemConfig = "/usr/lib/docker/config.json"
 )
 
 var (
@@ -124,6 +125,16 @@ func load(configDir string) (*configfile.ConfigFile, bool, error) {
 	filename := filepath.Join(configDir, ConfigFileName)
 	configFile := configfile.New(filename)
 
+   // LmP: Load values from system config by default
+   if _, err := os.Stat(defaultSystemConfig); err == nil {
+       file, err := os.Open(defaultSystemConfig)
+       if err == nil {
+           configFile.LoadFromReader(file)
+           file.Close()
+       }
+   }
+
+
 	// Try happy path first - latest config file
 	if file, err := os.Open(filename); err == nil {
 		defer file.Close()
