diff --git a/cli/config/config.go b/cli/config/config.go
index b7c05c3f86..1f56eb11ba 100644
--- a/cli/config/config.go
+++ b/cli/config/config.go
@@ -21,6 +21,7 @@ const (
 	configFileDir  = ".docker"
 	oldConfigfile  = ".dockercfg" // Deprecated: remove once we stop printing deprecation warning
 	contextsDir    = "contexts"
+	defaultSystemConfig = "/usr/lib/docker/config.json"
 )
 
 var (
@@ -114,6 +115,15 @@ func load(configDir string) (*configfile.ConfigFile, bool, error) {
 	filename := filepath.Join(configDir, ConfigFileName)
 	configFile := configfile.New(filename)
 
+	// LmP: Load values from system config by default
+	if _, err := os.Stat(defaultSystemConfig); err == nil {
+		file, err := os.Open(defaultSystemConfig)
+		if err == nil {
+			configFile.LoadFromReader(file)
+			file.Close()
+		}
+	}
+
 	// Try happy path first - latest config file
 	if file, err := os.Open(filename); err == nil {
 		defer file.Close()
