From e915a809d0f03e0f2ebb3ad585b2206c0d4b2511 Mon Sep 17 00:00:00 2001
From: Jorge Ramirez-Ortiz <jorge@foundries.io>
Date: Sun, 27 Oct 2019 16:39:45 +0100
Subject: [FIO toup] plat-imx: configure the SHMEM section

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>
---
 core/arch/arm/plat-imx/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/arch/arm/plat-imx/main.c b/core/arch/arm/plat-imx/main.c
index c632fa85..602365b5 100644
--- a/core/arch/arm/plat-imx/main.c
+++ b/core/arch/arm/plat-imx/main.c
@@ -112,6 +112,10 @@ register_phys_mem(MEM_AREA_TEE_COHERENT,
 		  CORE_MMU_PGDIR_SIZE);
 #endif
 
+#ifdef TEE_SHMEM_START
+register_dynamic_shm(TEE_SHMEM_START, TEE_SHMEM_SIZE);
+#endif
+
 #if defined(CFG_PL310)
 register_phys_mem_pgdir(MEM_AREA_IO_SEC,
 			ROUNDDOWN(PL310_BASE, CORE_MMU_PGDIR_SIZE),
-- 
2.17.1

