Allow custom bin dir and lib via ${BINDIR}/${LIBDIR}.

diff --git a/Makefile b/Makefile
index 7a0018a..ccb7997 100644
--- a/Makefile
+++ b/Makefile
@@ -211,9 +211,9 @@ patch:
 endif
 
 install:
-	$(echo) '  INSTALL ${DESTDIR}/lib/optee_armtz'
-	$(q)mkdir -p ${DESTDIR}/lib/optee_armtz
-	$(q)find $(out-dir) -name \*.ta -exec cp -a {} ${DESTDIR}/lib/optee_armtz \;
-	$(echo) '  INSTALL ${DESTDIR}/bin'
-	$(q)mkdir -p ${DESTDIR}/bin
-	$(q)cp -a $(out-dir)/xtest/xtest ${DESTDIR}/bin
+	$(echo) '  INSTALL ${DESTDIR}/${LIBDIR}/optee_armtz'
+	$(q)mkdir -p ${DESTDIR}/${LIBDIR}/optee_armtz
+	$(q)find $(out-dir) -name \*.ta -exec cp -a {} ${DESTDIR}/${LIBDIR}/optee_armtz \;
+	$(echo) '  INSTALL ${DESTDIR}/${BINDIR}'
+	$(q)mkdir -p ${DESTDIR}/${BINDIR}
+	$(q)cp -a $(out-dir)/xtest/xtest ${DESTDIR}/${BINDIR}
