Set default fusemaps directory to /usr/share/crucible/fusemaps.

Upstream-Status: Inappropriate [OE specific]

diff --git a/src/github.com/inversepath/crucible/crucible.go b/src/github.com/inversepath/crucible/crucible.go
index a633df8..8269e9a 100644
--- a/src/github.com/inversepath/crucible/crucible.go
+++ b/src/github.com/inversepath/crucible/crucible.go
@@ -13,13 +13,11 @@ import (
 	"errors"
 	"flag"
 	"fmt"
-	"go/build"
 	"io/ioutil"
 	"log"
 	"log/syslog"
 	"math/big"
 	"os"
-	"path"
 	"path/filepath"
 	"strings"
 	"text/tabwriter"
@@ -104,7 +102,7 @@ func init() {
 	flag.StringVar(&conf.reference, "r", "", "reference manual revision")
 
 	if conf.fusemaps == "" {
-		conf.fusemaps = path.Join(build.Default.GOPATH, "src/github.com/inversepath/crucible/fusemaps")
+		conf.fusemaps = "/usr/share/crucible/fusemaps"
 	}
 }
 
