Enable standard distro configs by default.

Index: git/include/configs/cl-som-imx7.h
===================================================================
--- git.orig/include/configs/cl-som-imx7.h
+++ git/include/configs/cl-som-imx7.h
@@ -74,6 +74,8 @@
 #define CONFIG_BOOTDELAY		3
 #define CONFIG_SYS_AUTOLOAD		"no"
 
+#include <config_distro_defaults.h>
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"autoload=off\0" \
 	"script=boot.scr\0" \
