Index: git/include/configs/cl-som-imx7.h
===================================================================
--- git.orig/include/configs/cl-som-imx7.h
+++ git/include/configs/cl-som-imx7.h
@@ -82,21 +82,23 @@
 	"loadscript=load ${storagetype} ${storagedev} ${loadaddr} ${script};\0" \
 	"loadkernel=load ${storagetype} ${storagedev} ${loadaddr} ${kernel};\0" \
 	"loadkernel_nand=nand read ${loadaddr} 0 ${kernel_size}\0" \
-	"loadfdt=load ${storagetype} ${storagedev} ${fdtaddr} ${fdtfile};\0" \
-	"loadfdt_nand=nand read ${fdtaddr} 980000 10000\0" \
+	"loadfdt=load ${storagetype} ${storagedev} ${fdt_addr_r} ${fdtfile};\0" \
+	"loadfdt_nand=nand read ${fdt_addr_r} 980000 10000\0" \
 	"bootscript=echo Running bootscript from ${storagetype} ...; source ${loadaddr};\0" \
 	"storagebootcmd=echo Booting from ${storagetype} ...; run ${storagetype}args; run doboot;\0" \
 	"kernel=zImage\0" \
 	"kernel_size=800000\0" \
 	"console=ttymxc0\0" \
+	"fdt_addr_r=0x82000000\0" \
+	"kernel_addr_r=0x81000000\0" \
+	"ramdisk_addr_r=0x82100000\0" \
 	"fdt_high=0xffffffff\0" \
 	"initrd_high=0xffffffff\0" \
-	"fdtaddr=0x83000000\0" \
 	"mmcdev_def="__stringify(CONFIG_SYS_MMC_DEV)"\0" \
 	"usbdev_def="__stringify(CONFIG_SYS_USB_DEV)"\0" \
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
 	"usbpart=" __stringify(CONFIG_SYS_USB_IMG_LOAD_PART) "\0" \
-	"doboot=bootz ${loadaddr} - ${fdtaddr}\0" \
+	"doboot=bootz ${loadaddr} - ${fdt_addr_r}\0" \
 	"mmc_config=mmc dev ${mmcdev}; mmc rescan\0" \
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
 		"root=/dev/mmcblk${mmcblk}p2 rootwait rw\0" \
