diff --git a/secure_storage/Makefile b/secure_storage/Makefile
index 3b9a8dc..dfa4f8b 100644
--- a/secure_storage/Makefile
+++ b/secure_storage/Makefile
@@ -6,8 +6,8 @@ TA_CROSS_COMPILE ?= $(CROSS_COMPILE)
 
 .PHONY: all
 all:
-	$(MAKE) -C host CROSS_COMPILE="$(HOST_CROSS_COMPILE)"
-	$(MAKE) -C ta CROSS_COMPILE="$(TA_CROSS_COMPILE)"
+	$(MAKE) -C host CROSS_COMPILE="$(HOST_CROSS_COMPILE)" --no-builtin-variables
+	$(MAKE) -C ta CROSS_COMPILE="$(TA_CROSS_COMPILE)" LDFLAGS=""
 
 .PHONY: clean
 clean:
diff --git a/secure_storage/host/Makefile b/secure_storage/host/Makefile
index 5e2c801..29bfb87 100644
--- a/secure_storage/host/Makefile
+++ b/secure_storage/host/Makefile
@@ -1,10 +1,10 @@
-CC      = $(CROSS_COMPILE)gcc
-LD      = $(CROSS_COMPILE)ld
-AR      = $(CROSS_COMPILE)ar
-NM      = $(CROSS_COMPILE)nm
-OBJCOPY = $(CROSS_COMPILE)objcopy
-OBJDUMP = $(CROSS_COMPILE)objdump
-READELF = $(CROSS_COMPILE)readelf
+CC      ?= $(CROSS_COMPILE)gcc
+LD      ?= $(CROSS_COMPILE)ld
+AR      ?= $(CROSS_COMPILE)ar
+NM      ?= $(CROSS_COMPILE)nm
+OBJCOPY ?= $(CROSS_COMPILE)objcopy
+OBJDUMP ?= $(CROSS_COMPILE)objdump
+READELF ?= $(CROSS_COMPILE)readelf
 
 OBJS = main.o
 
@@ -23,3 +23,6 @@ $(BINARY): $(OBJS)
 .PHONY: clean
 clean:
 	rm -f $(OBJS) $(BINARY)
+
+%.o: %.c
+	$(CC) $(CFLAGS) -c $< -o $@
