From a94945325466109d9723b520b2fee1ce9370f1d9 Mon Sep 17 00:00:00 2001
From: Tim Anderson <tim.anderson@foundries.io>
Date: Fri, 17 Jun 2022 16:27:45 +0000
Subject: [PATCH] am62xx-evm: set bootm len to 64M

Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
---
 include/configs/am62x_evm.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/am62x_evm.h b/include/configs/am62x_evm.h
index eb7c2a287c..6ae3f4ad34 100644
--- a/include/configs/am62x_evm.h
+++ b/include/configs/am62x_evm.h
@@ -50,6 +50,8 @@
 #define CONFIG_SYS_SPL_MALLOC_SIZE     SZ_16M
 #endif
 
+#define CONFIG_SYS_BOOTM_LEN            SZ_64M
+
 #define PARTS_DEFAULT \
 	/* Linux partitions */ \
 	"name=rootfs,start=0,size=-,uuid=${uuid_gpt_rootfs}\0"
-- 
2.25.1

