From 1c0e37ef9eb8f25afae58007c947ff34afbff41f Mon Sep 17 00:00:00 2001
From: Tyler Baker <tyler@foundries.io>
Date: Tue, 29 Sep 2020 21:35:28 -0700
Subject: [PATCH] include/configs/stm32mp1: set lmp boot cmd

Signed-off-by: Tyler Baker <tyler@foundries.io>
---
 include/configs/stm32mp1.h | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/include/configs/stm32mp1.h b/include/configs/stm32mp1.h
index 7b4fe484d3..46ce822435 100644
--- a/include/configs/stm32mp1.h
+++ b/include/configs/stm32mp1.h
@@ -138,21 +138,14 @@
  */
 
 #define STM32MP_BOOTCMD "bootcmd_stm32mp=" \
-	"echo \"Boot over ${boot_device}${boot_instance}!\";" \
 	"if test ${boot_device} = serial || test ${boot_device} = usb;" \
 	"then stm32prog ${boot_device} ${boot_instance}; " \
 	"else " \
-		"run env_check;" \
-		"if test ${boot_device} = mmc;" \
-		"then env set boot_targets \"mmc${boot_instance}\"; fi;" \
-		"if test ${boot_device} = nand ||" \
-		  " test ${boot_device} = spi-nand ;" \
-		"then env set boot_targets ubifs0; fi;" \
-		"if test ${boot_device} = nor;" \
-		"then env set boot_targets mmc0; fi;" \
-		"run distro_bootcmd;" \
+		"setenv devtype mmc ; setenv devnum 1 ; setenv fdtfile stm32mp157c-ev1.dtb ; setenv addr_fit 0xC0008000 ;" \
+		"load ${devtype} ${devnum}:2 ${addr_fit} /boot/loader/uEnv.txt ; env import -t ${addr_fit} ${filesize} ;" \
+		"load ${devtype} ${devnum}:2 ${addr_fit} \"/boot\"${kernel_image} ; bootm ${addr_fit}#conf@${fdtfile} ;" \
 	"fi;\0"
-
+	
 /* DTIMG command added only for Android distribution */
 #ifdef CONFIG_CMD_DTIMG
 /*
-- 
2.26.0
