diff --git a/resize-helper b/resize-helper
index cfa9ab7..e353f34 100755
--- a/resize-helper
+++ b/resize-helper
@@ -48,6 +48,8 @@ if [ "$PART_TABLE_TYPE" = "gpt" ]; then
 	${PARTPROBE}
 fi
 
-${PARTED} ${DEVICE} resizepart ${PART_ENTRY_NUMBER} 100%
+# 'Yes' is required when partition is already mounted
+${PARTED} -s ${DEVICE} resizepart ${PART_ENTRY_NUMBER} 100% || \
+	${PARTED} ---pretend-input-tty ${DEVICE} resizepart ${PART_ENTRY_NUMBER} Yes 100%
 ${PARTPROBE}
 ${RESIZE2FS} "${ROOT_DEVICE}"
