From 2f72e2c64b152b28b5620c1baa46d38bd0addd5e Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Fri, 19 Jul 2019 20:13:24 -0300
Subject: [PATCH] rpi4: prefer downstream dtb

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 board/raspberrypi/rpi/rpi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
index 20c77a3554..b685e9d0d0 100644
--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -150,7 +150,7 @@ static const struct rpi_model rpi_models_new_scheme[] = {
 	},
 	[0x11] = {
 		"4 Model B",
-		DTB_DIR "bcm2838-rpi-4-b.dtb",
+		DTB_DIR "bcm2711-rpi-4-b.dtb",
 		true,
 	},
 };
-- 
2.22.0

