From 6f900c882a4abed38cfdcbd4d4257bb3f61b85b4 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@opensourcefoundries.com>
Date: Thu, 7 Jun 2018 19:22:27 -0300
Subject: [PATCH] configs/dragonboard820: remove qcom dir from fdtfile

OE installs the device tree at the vfat root.

Signed-off-by: Ricardo Salveti <ricardo@opensourcefoundries.com>

diff --git a/include/configs/dragonboard820c.h b/include/configs/dragonboard820c.h
index 12ae6bc..b0d33dd 100644
--- a/include/configs/dragonboard820c.h
+++ b/include/configs/dragonboard820c.h
@@ -47,7 +47,7 @@
 	"initrd_high=0xffffffffffffffff\0" \
 	"linux_image=uImage\0" \
 	"kernel_addr_r=0x95000000\0"\
-	"fdtfile=qcom/apq8096-db820c.dtb\0" \
+	"fdtfile=apq8096-db820c.dtb\0" \
 	"fdt_addr_r=0x93000000\0"\
 	"ramdisk_addr_r=0x91000000\0"\
 	"scriptaddr=0x90000000\0"\
-- 
2.7.4

