From c8c4fcdf04bebcb9593441003a51cda07b022495 Mon Sep 17 00:00:00 2001
From: Andy Doan <andy@foundries.io>
Date: Wed, 14 Aug 2019 14:12:37 -0500
Subject: [PATCH 06/14] aktualizr-lite: Fix incorrect storing of ecu hwid

Signed-off-by: Andy Doan <andy@foundries.io>
---
 src/aktualizr_lite/main.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/aktualizr_lite/main.cc b/src/aktualizr_lite/main.cc
index 9bc6efad..9b544a50 100644
--- a/src/aktualizr_lite/main.cc
+++ b/src/aktualizr_lite/main.cc
@@ -56,7 +56,7 @@ static std::shared_ptr<SotaUptaneClient> liteClient(Config &config, std::shared_
       boost::uuids::uuid tmp = boost::uuids::random_generator()();
       serial = boost::uuids::to_string(tmp);
     }
-    ecu_serials.emplace_back(Uptane::EcuSerial(serial), Uptane::HardwareIdentifier(serial));
+    ecu_serials.emplace_back(Uptane::EcuSerial(serial), Uptane::HardwareIdentifier(hwid));
     storage->storeEcuSerials(ecu_serials);
   }
 
-- 
2.23.0

