From 877e6602dd474902e9c5bdf5aa6c29b04d5ee3b7 Mon Sep 17 00:00:00 2001
From: Andy Doan <andy@foundries.io>
Date: Wed, 3 Jul 2019 23:10:04 -0500
Subject: [PATCH 11/12] aktualizr-lite: Add ability to reboot after an update

Signed-off-by: Andy Doan <andy@foundries.io>
---
 src/aktualizr_lite/main.cc | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/aktualizr_lite/main.cc b/src/aktualizr_lite/main.cc
index aa8f616e..b7fcb56b 100644
--- a/src/aktualizr_lite/main.cc
+++ b/src/aktualizr_lite/main.cc
@@ -263,6 +263,11 @@ static int update_main(Config &config, const bpo::variables_map &variables_map)
 }
 
 static int daemon_main(Config &config, const bpo::variables_map &variables_map) {
+  if (access(config.bootloader.reboot_command.c_str(), X_OK) != 0) {
+    LOG_ERROR << "reboot command: " << config.bootloader.reboot_command << " is not executable";
+    return 1;
+  }
+
   auto storage = INvStorage::newStorage(config.storage);
   auto client = liteClient(config, storage);
   bool compareDockerApps = should_compare_docker_apps(config);
@@ -285,7 +290,10 @@ static int daemon_main(Config &config, const bpo::variables_map &variables_map)
     if (target != nullptr && !targets_eq(*target, current, compareDockerApps)) {
       LOG_INFO << "Updating base image to: " << *target;
       if (do_update(*client, *storage, *target) == 0) {
-        // TODO reboot
+        if (std::system(config.bootloader.reboot_command.c_str()) != 0) {
+          LOG_ERROR << "Unable to reboot system";
+          return 1;
+        }
       }
     }
     sleep(interval);
-- 
2.23.0

