From 78aedb43b80e699f7eaf0c48aa7a6dd5e7afea5a Mon Sep 17 00:00:00 2001
From: Andy Doan <andy@foundries.io>
Date: Wed, 3 Jul 2019 23:10:04 -0500
Subject: [PATCH 13/14] aktualizr-lite: Add ability to reboot after an update

Signed-off-by: Andy Doan <andy@foundries.io>
---
 src/aktualizr_lite/main.cc | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/src/aktualizr_lite/main.cc b/src/aktualizr_lite/main.cc
index dfab5020..06a50a7b 100644
--- a/src/aktualizr_lite/main.cc
+++ b/src/aktualizr_lite/main.cc
@@ -257,6 +257,12 @@ static int update_main(Config &config, const bpo::variables_map &variables_map)
 }
 
 static int daemon_main(Config &config, const bpo::variables_map &variables_map) {
+  auto rebootCmd = variables_map["reboot-command"].as<boost::filesystem::path>();
+  if (access(rebootCmd.c_str(), X_OK) != 0) {
+    LOG_ERROR << "reboot command: " << rebootCmd << " is not executable";
+    return 1;
+  }
+
   auto storage = INvStorage::newStorage(config.storage);
   auto client = liteClient(config, storage);
   bool compareDockerApps = should_compare_docker_apps(config);
@@ -279,7 +285,10 @@ static int daemon_main(Config &config, const bpo::variables_map &variables_map)
     if (target != nullptr && !targets_eq(*target, current, compareDockerApps)) {
       LOG_INFO << "Updating base image to: " << *target;
       if (do_update(*client, *storage, *target) == 0) {
-        // TODO reboot
+        if (std::system(rebootCmd.c_str()) != 0) {
+          LOG_ERROR << "Unable to reboot system";
+          return 1;
+        }
       }
     }
     sleep(interval);
@@ -303,6 +312,11 @@ void check_info_options(const bpo::options_description &description, const bpo::
     std::cout << description << '\n';
     exit(EXIT_SUCCESS);
   }
+  if (vm["command"].as<std::string>() == "daemon" && vm.count("reboot-command") == 0) {
+    std::cout << "ERROR: --reboot-command required when running as daemon\n";
+    std::cout << description << '\n';
+    exit(EXIT_SUCCESS);
+  }
   if (vm.count("version") != 0) {
     std::cout << "Current aktualizr version is: " << aktualizr_version() << "\n";
     exit(EXIT_SUCCESS);
@@ -331,6 +345,7 @@ bpo::variables_map parse_options(int argc, char *argv[]) {
       ("primary-ecu-hardware-id", bpo::value<std::string>(), "hardware ID of primary ecu")
       ("update-name", bpo::value<std::string>(), "optional name of the update when running \"update\". default=latest")
       ("interval", bpo::value<unsigned int>(), "optional interval in seconds to poll for update when in daemon mode. default=300")
+      ("reboot-command", bpo::value<boost::filesystem::path>(), "reboot command to call after an update is applied from daemon mode")
       ("command", bpo::value<std::string>(), subs.c_str());
   // clang-format on
 
-- 
2.23.0

