diff --git a/acipher/host/Makefile b/acipher/host/Makefile
index 8f4bc8a..b471aab 100644
--- a/acipher/host/Makefile
+++ b/acipher/host/Makefile
@@ -10,7 +10,7 @@ OBJS = main.o
 
 CFLAGS += -Wall -I../ta/include -I./include
 CFLAGS += -I$(TEEC_EXPORT)/include
-LDADD += -lteec -L$(TEEC_EXPORT)/lib
+LDADD += $(LDFLAGS) -lteec -L$(TEEC_EXPORT)/lib
 
 BINARY = optee_example_acipher
 
diff --git a/aes/host/Makefile b/aes/host/Makefile
index dfeb4e8..1ea0e15 100644
--- a/aes/host/Makefile
+++ b/aes/host/Makefile
@@ -10,7 +10,7 @@ OBJS = main.o
 
 CFLAGS += -Wall -I../ta/include -I./include
 CFLAGS += -I$(TEEC_EXPORT)/include
-LDADD += -lteec -L$(TEEC_EXPORT)/lib
+LDADD += $(LDFLAGS) -lteec -L$(TEEC_EXPORT)/lib
 
 BINARY = optee_example_aes
 
diff --git a/hello_world/host/Makefile b/hello_world/host/Makefile
index c4c8239..aac5063 100644
--- a/hello_world/host/Makefile
+++ b/hello_world/host/Makefile
@@ -10,7 +10,7 @@ OBJS = main.o
 
 CFLAGS += -Wall -I../ta/include -I$(TEEC_EXPORT)/include -I./include
 #Add/link other required libraries here
-LDADD += -lteec -L$(TEEC_EXPORT)/lib
+LDADD += $(LDFLAGS) -lteec -L$(TEEC_EXPORT)/lib
 
 BINARY = optee_example_hello_world
 
diff --git a/hotp/host/Makefile b/hotp/host/Makefile
index cb7fd19..2dc6f95 100644
--- a/hotp/host/Makefile
+++ b/hotp/host/Makefile
@@ -10,7 +10,7 @@ OBJS = main.o
 
 CFLAGS += -Wall -I../ta/include -I./include
 CFLAGS += -I$(TEEC_EXPORT)/include
-LDADD += -lteec -L$(TEEC_EXPORT)/lib
+LDADD += $(LDFLAGS) -lteec -L$(TEEC_EXPORT)/lib
 
 BINARY = optee_example_hotp
 
diff --git a/random/host/Makefile b/random/host/Makefile
index fd407d9..3e2fc40 100644
--- a/random/host/Makefile
+++ b/random/host/Makefile
@@ -10,7 +10,7 @@ OBJS = main.o
 
 CFLAGS += -Wall -I../ta/include -I./include
 CFLAGS += -I$(TEEC_EXPORT)/include
-LDADD += -lteec -L$(TEEC_EXPORT)/lib
+LDADD += $(LDFLAGS) -lteec -L$(TEEC_EXPORT)/lib
 
 BINARY = optee_example_random
 
diff --git a/secure_storage/host/Makefile b/secure_storage/host/Makefile
index 29bfb87..debc33c 100644
--- a/secure_storage/host/Makefile
+++ b/secure_storage/host/Makefile
@@ -10,7 +10,7 @@ OBJS = main.o
 
 CFLAGS += -Wall -I../ta/include -I./include
 CFLAGS += -I$(TEEC_EXPORT)/include
-LDADD += -lteec -L$(TEEC_EXPORT)/lib
+LDADD += $(LDFLAGS) -lteec -L$(TEEC_EXPORT)/lib
 
 BINARY = optee_example_secure_storage
 
