diff --git a/src/Makefile b/src/Makefile
index 917bf6b..9a02d4f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -20,7 +20,7 @@ export DATADIR
 export SQLITE
 export ZLIB
 
-CXX?=c++
+CXX?=$(CROSS_COMPILE)c++
 INCLUDES=-I./core/
 DEFINES=-DPREFIX=\"$(PREFIX)\" -DSBINDIR=\"$(SBINDIR)\" -DMANDIR=\"$(MANDIR)\" -DDATADIR=\"$(DATADIR)\" -DVERSION=\"$(VERSION)\"
 CXXFLAGS=-g -Wall -g $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
diff --git a/src/core/Makefile b/src/core/Makefile
index 884a963..fbf6027 100644
--- a/src/core/Makefile
+++ b/src/core/Makefile
@@ -1,6 +1,6 @@
 PACKAGENAME?=lshw
 
-CXX=c++
+CXX?=$(CROSS_COMPILE)c++
 INCLUDES=
 DEFINES=-DPREFIX=\"$(PREFIX)\" -DSBINDIR=\"$(SBINDIR)\" -DMANDIR=\"$(MANDIR)\" -DDATADIR=\"$(DATADIR)\"
 CXXFLAGS?=-g -Wall $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
diff --git a/src/gui/Makefile b/src/gui/Makefile
index 2526c18..74be693 100644
--- a/src/gui/Makefile
+++ b/src/gui/Makefile
@@ -1,7 +1,7 @@
 PACKAGENAME?=lshw
 
-CXX?=c++
-CC?=cc
+CXX?=$(CROSS_COMPILE)c++
+CC?=$(CROSS_COMPILE)cc
 STRIP?=strip
 OBJCOPY?=objcopy
 
