diff --git a/src/Makefile b/src/Makefile
index 917bf6b..4766fcb 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -30,9 +30,9 @@ endif
 ifeq ($(ZLIB), 1)
 	CXXFLAGS+= -DZLIB $(shell pkg-config --cflags zlib)
 endif
-LDFLAGS=-L./core/ -g
+LDEXTRAS=-L./core/ -g
 ifneq ($(shell $(LD) --help 2| grep -- --as-needed), )
-	LDFLAGS+= -Wl,--as-needed
+	LDEXTRAS+= -Wl,--as-needed
 endif
 LDSTATIC=-static
 LIBS+=-llshw -lresolv
@@ -45,7 +45,7 @@ endif
 
 export CXXFLAGS
 export LIBS
-export LDFLAGS
+export LDEXTRAS
 
 ifeq ($(ZLIB), 1)
 DATAFILES = pci.ids.gz usb.ids.gz oui.txt.gz manuf.txt.gz pnp.ids.gz pnpid.txt.gz
@@ -66,7 +66,7 @@ core:
 	+make -C core all
 
 $(PACKAGENAME): core $(PACKAGENAME).o
-	$(CXX) $(LDFLAGS) -o $@ $(PACKAGENAME).o $(LIBS)
+	$(CXX) $(LDFLAGS) ${LDEXTRAS} -o $@ $(PACKAGENAME).o $(LIBS)
 
 .PHONY: po
 po:
@@ -84,7 +84,7 @@ nologo:
 static: $(PACKAGENAME)-static
 
 $(PACKAGENAME)-static: core core/lib$(PACKAGENAME).a $(PACKAGENAME).o
-	$(CXX) $(LDSTATIC) $(LDFLAGS) -o $@ $(PACKAGENAME).o $(LIBS)
+	$(CXX) $(LDSTATIC) $(LDFLAGS) ${LDEXTRAS} -o $@ $(PACKAGENAME).o $(LIBS)
 	$(STRIP) $@
 
 .PHONY: compressed
