diff --git a/finish b/finish
index 717383e..f34edd8 100755
--- a/finish
+++ b/finish
@@ -14,10 +14,11 @@ finish_run() {
 
 		info "Switching root to '$ROOTFS_DIR'..."
 
-		debug "Moving /dev, /proc and /sys onto rootfs..."
+		debug "Moving /dev, /proc, /sys and /run onto rootfs..."
 		mount --move /dev $ROOTFS_DIR/dev
 		mount --move /proc $ROOTFS_DIR/proc
 		mount --move /sys $ROOTFS_DIR/sys
+		mount --move /run $ROOTFS_DIR/run
 
 		cd $ROOTFS_DIR
 		exec switch_root -c /dev/console $ROOTFS_DIR ${bootparam_init:-/sbin/init}
diff --git a/init b/init
index c71ce0c..79cf5c1 100755
--- a/init
+++ b/init
@@ -78,9 +78,11 @@ EFI_DIR=/sys/firmware/efi  # place to store device firmware information
 touch /etc/fstab
 
 # initialize /proc, /sys, /run/lock and /var/lock
-mkdir -p /proc /sys /run/lock /var/lock
+mkdir -p /proc /sys /run /var/lock
 mount -t proc proc /proc
 mount -t sysfs sysfs /sys
+mount -t tmpfs tmpfs /run
+mkdir -p /run/lock
 
 if [ -d $EFI_DIR ];then
 	mount -t efivarfs none /sys/firmware/efi/efivars
